### PR TITLE
Update mixin_columns docs to reflect FITS and HDF5 writing

### PR DESCRIPTION
**Pull request by taldcroft**
This is a minor doc update as a follow-on to #7007 and #6912.

### CODE REVIEW

**Comment by taldcroft** [comment]
@bsipocz - can you read the little bit of  new text and see if it makes sense to you?

File: docs/table/mixin_columns.rst [hunk 1]
```diff
@@ -238,10 +238,19 @@ that contain mixin columns:
...
-writers will be used.
+writers will be used.  For writing tables with mixin columns it is recommended
+to use the ``'ecsv'`` ASCII format.  This will fully serialize the table data and
+metadata, allowing full "round-trip" of the table when it is read back.  See
+:ref:`ecsv_format` for details.
 
+**Binary table writing**
+
+Starting with astropy 3.0, tables with mixin columns can be written in binary
+format to file using the both FITS and HDF5 formats.  These can be read back to
```
Comment (bsipocz) [inline docs/table/mixin_columns.rst]: typo -> "using both FITS..."

File: docs/table/mixin_columns.rst [hunk 2]
```diff
@@ -238,10 +238,19 @@ that contain mixin columns:
...
+recover exactly the original table including mixin columns and metadata. See
```
Comment (bsipocz) [inline docs/table/mixin_columns.rst]: I think it will be a bit cleaner to capitalize Table here?

File: docs/table/mixin_columns.rst [hunk 3]
```diff
@@ -238,10 +238,19 @@ that contain mixin columns:
 
 **ASCII table writing**
 
-Mixin columns can be written out to file using the `astropy.io.ascii` module,
+Tables with mixin columns can be written out to file using the `astropy.io.ascii` module,
 but the fast C-based writers are not available.  Instead the legacy pure-Python
```
Comment (bsipocz) [inline docs/table/mixin_columns.rst]: I recall some discussion of not calling the pure-Python version legacy? Or was that just the opposite, to call it legacy? Sadly I couldn't find the issue now.
The former would make more sense to me, as the fast writer doesn't provide all the functionality.

Comment (taldcroft) [reply docs/table/mixin_columns.rst]: Good catch, the pure-Python writers are definitely not legacy.

**Pull request merged**